dolphinscheduler-ui/src/views/projects/workflow/definition/components/start-modal.tsx

```diff
@@ -505,16 +504,13 @@ export default defineComponent({
                     label={t('project.workflow.parallelism')}
                     path='expectedParallelismNumber'
                   >
-                    <NCheckbox v-model:checked={this.parallelismRef}>
-                      {t('project.workflow.custom_parallelism')}
-                    </NCheckbox>
-                    <NInput
-                      allowInput={this.trim}
-                      disabled={!this.parallelismRef}
+                    <NInputNumber
                       placeholder={t(
                         'project.workflow.please_enter_parallelism'
                       )}
                       v-model:value={this.startForm.expectedParallelismNumber}
+                      max='100'
+                      min='1'
```

Comment:
It's not a good idea to restrict the parallelism should between `1` and `100`, this shouldn't be restricted.

Reply:
I understand that an upper limit should be set. If there is no upper limit, all complement tasks will be triggered, which will put a lot of pressure on the system.

Reply:
I have encountered a real scenario where the user needs to supplement three years of data. After selecting the parallel mode, 1500+ workflow instances are generated at one time, which directly causes the dolphinscheduler worker service to go down. I suggest that there is an upper limit to limit it. We can discuss this upper limit. How much is appropriate to set.

Reply:
Master/Worker has its own protection strategy, I don't think we should rely on a workflow config to protect the system. And I don't think we can get the correct upper limit at this discussion, different user have different cluster size and usage Scenarios, if we don't know how to set the max value, why we need to set. In additional, if we want to avoid the system crash in a large parallel, the correct way is to limit the rate at server side.

Reply:
If an appropriate value cannot be negotiated, it should be returned by `api-server` and allow the administrator to configure it.

Reply:
It is a very good solution for Master/Worker to have its own protection strategy, but this part needs to be strengthened. At the same time, 1500+ complement workflows were submitted, I saw that the Worker service was directly down.

Reply:
You can try to set `task-execute-threads-full-policy` to be REJECT, then the worker will only receive 100 tasks.
https://github.com/apache/dolphinscheduler/blob/d1135eabc7df9358fd210e423785d1067af64698/dolphinscheduler-worker/src/main/resources/application.yaml#L66-L66

BTW, I agree with `Gallardot` if we want to set a maximal restriction, it's better to make this value to be configurable.

Reply:
> You can try to set `task-execute-threads-full-policy` to be REJECT, then the worker will only receive 100 tasks.
> 
> https://github.com/apache/dolphinscheduler/blob/d1135eabc7df9358fd210e423785d1067af64698/dolphinscheduler-worker/src/main/resources/application.yaml#L66-L66
> 
> BTW, I agree with @Gallardot if we want to set a maximal restriction, it's better to make this value to be configurable.

+1

Reply:
I'v improved the front-end parallelism strategy so that it won't be limited to a default max value.  It'll just offer a tip.
![image](https://github.com/apache/dolphinscheduler/assets/4928204/3e1d6453-aad1-43d7-b939-72530db22366)

Reply:
Because Master/Worker has its own protection strategy, giving user prompts is a solution.

Reply:
The warning message is too strange, why it's better not to be over 10?